pkg/clients/openshift/metadata.go

```diff
@@ -0,0 +1,59 @@
+package openshift
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	metadataConfigMap = "osd-cluster-metadata"
+	configNamespace   = "openshift-config"
+)
+
+// getOsdClusterMetadata returns osd-cluster-metadata configmap data array from openshift-config namespace
+// this contains metadata about the cluster
+func (c Client) getOsdClusterMetadata(ctx context.Context) (map[string]string, error) {
+	var cm corev1.ConfigMap
+	cmdata := new(map[string]string)
+	if err := c.Get(ctx, metadataConfigMap, configNamespace, &cm); err != nil {
+		return *cmdata, err
+	}
+	return cm.Data, nil
+}
+
+func (c Client) IsSts(ctx context.Context) (bool, error) {
+	var cmData map[string]string
+	var err error
+	if cmData, err = c.getOsdClusterMetadata(ctx); err != nil {
+		return false, err
+	}
```

Comment:
```suggestion
	cmData, err := c.getOsdClusterMetadata(ctx)
	if err != nil {
		return false, err
	}
```

all of these can be shortened by using the short variable declaration

Reply:
This is still not the direction we want to go. We don't need to have the global variables.

Each of these functions can be written like so

```
	cmData, err := c.getOsdClusterMetadata(ctx)
	if err != nil {
		return false, err
	}
```

you get the unresolved reference error because you are wrapping the call in the `if` condition which scopes the variables

Reply:
My approach is usually "if it is an err returned only, use the compact `if err := ...`". Rarely will I use it if there are multiple return values (unless one is being ignored like a map check `if _, ok := data["cluster_name"]; ...` for example) but I wouldn't say either approach is a hard and fast rule :+1: thanks for the update